New fuseholder package : 5x20mm pcb closed. Brand Schurter modele FAB

tk5ep opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/KiCad/kicad-packages3D?pullRequest=142) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/KiCad/kicad-packages3D?pullRequest=142) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/KiCad/kicad-packages3D?pullRequest=142) it.</sub>

poeschlr commented:
Does the footprint already exist for this model? (If not please add it as well if at all possible. This should be added to the 

For future pull requests please add a link to the datasheet and a screenshot of the part.
For reference i think this is the part: https://www.schurter.com/en/datasheet/FAB

The current (footprint) repo does a bad job of making it easy to find anything. We need a better naming convention
I would suggest something like this:
`Fuseholder_[Fuse Type]_[MAN]_[Manufacturer series name]_[MPN]{_Pad[pad length]{x[pad width]}}_[orientation]_[Closed | Open]{_footprint options}`

This would mean for this contribution a name similar to:

`Fuseholder_Cylinder-5x20mm_Schurter_FAB_[MPN]_Horizontal_Closed`
replace [MPN] with the part number (And make sure you spell Schurter correctly. You spelled it wrong and my contribution also has a spelling error.)

poeschlr commented:
After importing your step file i noticed that it is not fused.
And the "Body plates" exist multiple times. (The contacts)

Could you also add the wrl file? (use stepup as described in the [forum thread](https://forum.kicad.info/t/anyone-succesfully-using-sketchup-to-create-export-3d-shapes/8056) to export everything correctly.)

Could you also add your source files to the [packages3d-source](https://github.com/KiCad/kicad-packages3D-source) repo? (Freecad, catia, ...)

The origin for the footprint (and therefor for the 3d model) should be at pin 1.
To comply with other footprints pin two should then be towards the right (positive x direction).
This also complies to IPC standards (footprint orientation A)

tk5ep commented:
[Fuseholder5x20_Horizontal_closed_Schurter_FAB.zip](https://github.com/KiCad/kicad-packages3D/files/1388421/Fuseholder5x20_Horizontal_closed_Schurter_FAB.zip)
Well,

I'm little bit confused with all that... As i said, i'm beginning in all that stuff and it's a lot to learn... But i understand that there must by some rules... The actual libraries are a mess, and that's the main criticism i hear about Kicad !

1) yes this is the right datasheet. Where should i add it in the future ?
2) No, i couldn't find a footprint for that fusehlder, i had to do it.
3) In the 3d repo fuseholders, the naming is not the one you suggest. Are these files to be changed ? You're talking about the footprint naming, not the packages. Should they have the same name ?
The naming should be 
Fuseholder_Cylinder-5x20mm_Schurter_FAB_0031-3553_Horizontal_Closed
Rather long !
4) I reworked my Freecad file and can't see what's wrong with the pins. Anyway, i added some constraints. I send you my FreeCad file, just in case...
Now, when i try to use Stepup, my screen becomes blank when i click on the "make union" button...
5) i have NO IDEA what a fused design is nor how to change origin and orientation.

OK about the pin naming, i named the back pin 1 in my footprint, so that's OK.

Thanks and hats up for your support !

easyw commented:
Which is your FC version and OS?
Which is your StepUp version?
Have you selected the objects before click on make union button?

easyw commented:
I tried to open your FC file and logging the opening operation I get a lot of errors in Report View

> Log: Loading Sketcher module... done
> Log: Loading PartDesign module... done
> Log: Loading GUI of Sketcher module... done
> Log: Loading GUI of PartDesign module... done
> Log: File was not written with this version of the topology
> Err: not a dag exception in DAGView::Model::updateSlot()
> Err: not a dag exception in DAGView::Model::updateSlot()

So I would need to know your FC version to solve this riddle ;)

tk5ep commented:
Hi !

Let me redraw my package before digging into that...
It can be a mistake from my side.
Can you just tell lme how to move or set the origin to one place as asked ?
What is a fused design ?

yes i selected the parts. It only happened after i've some mods on the
design.

 OS: Windows 10
Word size of OS: 64-bit
Word size of FreeCAD: 64-bit
Version: 0.17.12396 (Git)
Build type: Release
Branch: master
Hash: fb2b9d6cef287d826a905b90928d41a03e38d928
Python version: 2.7.8
Qt version: 4.8.7
Coin version: 4.0.0a
OCC version: 7.1.0
Locale: French/France (fr_FR)

Steup is 7.1.5.2

Thanks

2017-10-16 19:20 GMT+02:00 easyw <notifications@github.com>:

> Which is your FC version and OS?
> Which is your StepUp version?
> Have you selected the objects before click on make union button?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/KiCad/kicad-packages3D/pull/142#issuecomment-336958497>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABlBhUro6JOqtjs8wJE1gQRYVSQTnTMLks5ss5BZgaJpZM4P6oCN>
> .
>



-- 
Patrick Egloff
email : pegloff@gmail.com
Web page : http://www.egloff.eu

easyw commented:
Saving singularly Pin2 and Pin1 and re-opening them no errors.
Saving the Body as single file and reopening it the errors will appear on Report View...
so just focus on Body 😄 

> What is a fused design ?

In Part Menu, Boolean and then Union will make a fusion of all objects

tk5ep commented:
OK,

Can you check this one ? Now that i have a bit more experience, i redraw
this newer version without too many trouble.. Took the opportunity to be a
bit more accurate to original design...
I exported it using StepUp and the export to Kicad button.

Now, i need to knwo how to set origin and orientation... 😇

Thanks a lot,

2017-10-16 20:53 GMT+02:00 easyw <notifications@github.com>:

> Saving singularly Pin2 and Pin1 and re-opening them no errors.
> Saving the Body as single file and reopening it the errors will appear on
> Report View...
> so just focus on Body 😄
>
> What is a fused design ?
>
> In Part Menu, Boolean and then Union will make a fusion of all objects
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/KiCad/kicad-packages3D/pull/142#issuecomment-336994323>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABlBhQyPPAOvSh1Pc1U5WqD8cLDO9kf6ks5ss6YpgaJpZM4P6oCN>
> .
>



-- 
Patrick Egloff
email : pegloff@gmail.com
Web page : http://www.egloff.eu

easyw commented:
May I have the footprint?

poeschlr commented:
Regarding naming convention:
Yes we are reworking this. The rules have been lacking in the past. This means some libs have quite bad footprint naming. The new KLC v3 will define better rules. The discussion about the footprint naming can be found here: https://github.com/KiCad/kicad-library/issues/1447#issuecomment-323785988 (and in my [fork of the kicad webside](https://github.com/poeschlr/kicad-website/tree/fp_naming_convention). I hope i will soon find the time to finish this.)

easyw commented:
You can use StepUp to load footprint in FC and then align the model to fp

tk5ep commented:
I tried to align the 3dshape and this footprint with StepUp, but it's
really not intuitive...

I saw that my footprint had bad orientation and scale... i corrected this...
One step further, but enough for tonight !



2017-10-16 21:16 GMT+02:00 easyw <notifications@github.com>:

> May I have the footprint?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/KiCad/kicad-packages3D/pull/142#issuecomment-337003302>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABlBhbpRJsOdyBiZA9A0--10-KEYhsqWks5ss6uSgaJpZM4P6oCN>
> .
>



-- 
Patrick Egloff
email : pegloff@gmail.com
Web page : http://www.egloff.eu

easyw commented:
You can use Assembly2 or even my new Manipulator WB to align your model to footprint, but if you don't attach the fp or the new FC file, I can't be of much help...

poeschlr commented:
The normal workflow for me is to make the footprint first, import it into freecad via stepup and then design the part such that it is already correctly aligned to this footprint.

The footprint could look similar to this: (It is the footprint i made a while back but never committed. I now made it KLC compliant.) Remove the txt ending and import it into the footprint editor. (github does not allow kicad_mod files in the comment)
[Fuseholder_Cylinder-5x20mm_Schurter_FAB_0031-355x_Horizontal_Closed.kicad_mod.txt](https://github.com/KiCad/kicad-packages3D/files/1390270/Fuseholder_Cylinder-5x20mm_Schurter_FAB_0031-355x_Horizontal_Closed.kicad_mod.txt)
(Just to clear it up. You don't need to use this footprint. I also do not guarantee that everything is correct. I did not use this component in my PCB but the bigger FUP version.)

Also it seems you added your coil to this pull request as well. Please make one pull request per part.
(For this to work each pull request needs to be connected to its own git branch.)

poeschlr commented:
@SchrodingersGat should we include a [THT | SMD] field at the end of these footprint names for easy identification what part is SMD and what part is THT?

poeschlr commented:
Footprint pull request: https://github.com/KiCad/Fuse_Holders_and_Fuses.pretty/pull/17

tk5ep commented:
![kf301-2](https://user-images.githubusercontent.com/1655173/31981754-793f9fdc-b955-11e7-83c3-e7601e73f847.PNG)

poeschlr commented:
Please make one pull request per topic. (So this pull request should be split into 3. one for the fuse holder, one for the coil and one for the terminal block.)

About the fuse holder:
the wrl and step files are on the correct place and have the correct scaling but both have the wrong name.

The wrl file has a dash instead of an underline between FAB and the part number.
The step file still has the old naming. 

The correct name would be:
`Fuseholder_Cylinder-5x20mm_Schurter_FAB_0031-355x_Horizontal_Closed`

poeschlr commented:
The fuse holder model is ready to merge. (It is a very nice model indeed.)
Could you please remove the other files (coil and terminal block) from this pull request? (make sure you make a local copy of them.) As i wrote above they should have their own pull request.

And remember to share your source files over at the [packages3d-source](https://github.com/KiCad/kicad-packages3D-source) repo.
It would also be nice if you could enter your information to the Credits file found in the 3dshapes directory where you add a model.

tk5ep commented:
Hi,

Excuse my ignorance, but how do i remove the PR ? I'm absolutely not familair with Github and i still have a lot to learn...
Same for the credit...
Will put the source in the repo you asked.

Thanks,

poeschlr commented:
You don't need to remove the Pull Request. You have added more files to this pull request than necessary. (I don't know how to do it in the web interface. But you already seem to have managed it for the wrongly named fuse models.)
![screenshot from 2017-11-02 18-47-55](https://user-images.githubusercontent.com/18327350/32341758-b7565cf2-bffe-11e7-8ab5-2f67b15381e8.png)

To remove these files simply delete them from your branch.
Just make sure you have a local copy of them before you remove them. (Makes it easier for you do create a new pull request for them.)


There is a [CREDITS.md](https://github.com/tk5ep/kicad-packages3D/blob/master/Fuse_Holders_and_Fuses.3dshapes/CREDITS.md) file in the same directory where you placed your 3d models. Use any editor (even the online editor is ok) to add the information about your model. (Model name, who created it, what software was used)
If you use the online editor you need to ensure you edit the file in your fork. (The link already points to your fork)

---

For the future simply remember: One branch per contribution. (do not use the master branch of your fork as a source for pull requests.)
I found this tutorial about the web interface: https://github.com/WGBH/pbucore/wiki/Contributing-to-the-project-through-Github-web-interface

This seems to be a nice guide if you are using a local client: (This one is about the [github desktop app](https://desktop.github.com/))
https://www.davepaquette.com/archive/2016/01/24/Submitting-Your-First-Pull-request.aspx

I really suggest you go the route of using some local client.

tk5ep commented:
René,

This GitHub is driving me crazy !
There is probably a good reason and logic behind all this, but for a newcomer and novice, it's really boring !
Ok to remove the files, but from what repo do you mean ? The inductors_SMD and connectors ?
It seems that these files are of no interest, so will delete them and not upload again.

Thanks

poeschlr commented:
I deliberately did not look at them because this is the pull request for the fuses. (Keeps the discussion clean.)
If these other models have a footprint then they might be good to have. (But they should be added in their own pull request.)

easyw commented:
@tk5ep 
> This GitHub is driving me crazy !

You are not alone!
Please have a look at this video 😄 
https://youtu.be/CDeG4S-mJts

tk5ep commented:
LOL,

Linus coded Git as a joke ! ;-)

Shackmeister commented:
hi @tk5ep 
what is the status of this PR?

tk5ep commented:
Hi,

For me it has been close...

2017-12-03 3:41 GMT+01:00 Frank Severinsen <notifications@github.com>:

> hi @tk5ep <https://github.com/tk5ep>
> what is the status of this PR?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/KiCad/kicad-packages3D/pull/142#issuecomment-348735933>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABlBhW6HW3yQHf5aUZUThTJXgg_CSi--ks5s8gplgaJpZM4P6oCN>
> .
>



-- 
Patrick Egloff
email : pegloff@gmail.com
Web page : http://www.egloff.eu

Shackmeister commented:
@tk5ep can you please close this PR then? :)
